cmd/helm/repo_update.go

```diff
@@ -116,15 +116,22 @@ func (o *repoUpdateOptions) run(out io.Writer) error {
 
 func updateCharts(repos []*repo.ChartRepository, out io.Writer, failOnRepoUpdateFail bool) error {
 	fmt.Fprintln(out, "Hang tight while we grab the latest from your chart repositories...")
-	var wg sync.WaitGroup
-	var repoFailList []string
+
+	var (
+		wg           sync.WaitGroup
+		mu           sync.Mutex
+		repoFailList []string
```

Comment:
Can we convert `repoFailList` to a channel here? Failures can then be appended concurrently to that:

```suggestion
		repoFailList  := make(chan string, len(repos))
```

Then below:
```
if _, err := re.DownloadIndexFile(); err != nil {
    ...
    repoFailList <- re.Config.URL
```

The error handling will need to be adjusted too. ie. to reading from the channel. But that is trivial.

Reply:
fixed